test/hotspot/jtreg/compiler/lib/compile_framework/CompileFramework.java

```diff
@@ -0,0 +1,169 @@
+    private final Path classesDir = Utils.makeUniqueDir("compile-framework-classes-");
+    private ClassLoader classLoader;
+
+    /**
+     * Set up a new Compile Framework instance, for a new compilation unit.
+     */
+    public CompileFramework() {}
+
+    /**
+     * Add a Java source to the compilation.
+     *
+     * @param className Class name of the class (e.g. "{@code p.xyz.YXZ}").
+     * @param code Java code for the class, in the form of a {@link String}.
+     */
+    public void addJavaSourceCode(String className, String code) {
+        javaSources.add(new SourceCode(className, "java", code));
+    }
+
+    /**
+     * Add a Jasm source to the compilation.
+     *
+     * @param className Class name of the class (e.g. "{@code p.xyz.YXZ}").
+     * @param code Jasm code for the class, in the form of a {@link String}.
+     */
+    public void addJasmSourceCode(String className, String code) {
+        jasmSources.add(new SourceCode(className, "jasm", code));
+    }
+
+    /**
+     * Compile all sources: store the sources to the {@link sourceDir} directory, compile
+     * Java and Jasm sources and store the generated class-files in the {@link classesDir}
+     * directory.
+     */
+    public void compile() {
+        if (classLoader != null) {
+            throw new CompileFrameworkException("Cannot compile twice!");
+        }
+
+        Utils.printlnVerbose("------------------ CompileFramework: SourceCode -------------------");
+        Utils.printlnVerbose(sourceCodesAsString(jasmSources));
+        Utils.printlnVerbose(sourceCodesAsString(javaSources));
+
+        System.out.println("------------------ CompileFramework: Compilation ------------------");
+        System.out.println("Source directory: " + sourceDir);
+        System.out.println("Classes directory: " + classesDir);
+
+        Compile.compileJasmSources(jasmSources, sourceDir, classesDir);
+        Compile.compileJavaSources(javaSources, sourceDir, classesDir);
+        classLoader = ClassLoaderBuilder.build(classesDir);
+    }
+
+    private static String sourceCodesAsString(List<SourceCode> sourceCodes) {
+        StringBuilder builder = new StringBuilder();
+        for (SourceCode sourceCode : sourceCodes) {
+            builder.append("SourceCode: ").append(sourceCode.filePathName()).append(System.lineSeparator());
+            builder.append(sourceCode.code()).append(System.lineSeparator());
+        }
+        return builder.toString();
+    }
+
+    /**
+     * Access a class from the compiled code.
+     *
+     * @param name Name of the class to be retrieved.
+     * @return A class corresponding to the {@code name}.
+     */
+    public Class<?> getClass(String name) {
+        try {
+            return Class.forName(name, true, classLoader);
+        } catch (ClassNotFoundException e) {
+            throw new CompileFrameworkException("Class not found:", e);
+        }
+    }
+
+    /**
+     * Invoke a static method from the compiled code.
```

Comment:
Maybe add a new line for separation:
```suggestion
     * Invoke a static method from the compiled code.
     
```